add rockspec

teto opened the pull request:
- add rockspec: sqlite is a dependency but I chose not to expose it
- hack to fix tests: wouldn't it be simpler to use the envvar in all cases ?

tami5 commented:
Currently the ci tests failing for two reason: 

- upper/lower type names: Had this issue for awhile, I guess we need to make sure that is "always" lower case.
- test/auto/stmt_spec.lua:96: types no longer return in ci, maybe this should just be ignored/commented out

tami5 commented:
> I tried to make `luarocks test` succeeds and it was failing because of that code: sqlite_clib_path is not set in the minimal_test.vim. 

In that case, how about if not vim, try to read path from env var? or regardless if not set then read from env var. That way users may have the freedom to pick either one. 

> unrelated but would also be nice to have a vim.health script to check sqlite is correctly loaded.

Definitely it would, can you open an issue so we can later implement

teto commented:
this should be more acceptable. 

>    upper/lower type names: Had this issue for awhile, I guess we need to make sure that is "always" lower case.
>    test/auto/stmt_spec.lua:96: types no longer return in ci, maybe this should just be ignored/commented out

sry I didn't check those, could you rerun the workflow so I have a look please ?

tami5 commented:
LGTM, I will fix test later, in another pr along with ci issues

teto commented:
I've added a makefile target to run stylua locally so that I could fix the issues. The problem is that the stylua version I use locally may be different than the one on the CI. Since you are a nix user, would you mind if I added a flake just for development ? to get lualint/stylint and correctly export LIBSQLITE ?

tami5 commented:
> I've added a makefile target to run stylua locally so that I could fix the issues. The problem is that the stylua version I use locally may be different than the one on the CI. Since you are a nix user, would you mind if I added a flake just for development ? to get lualint/stylint and correctly export LIBSQLITE ?

Not at all. I like where this is going.

teto commented:
I've added the flake, running `nix develop` followed by `luarocks test` seems to pass. I wouldn't mind this being merged if only so that the github workflows get run automatically in followup PRs ^^'

tami5 commented on flake.nix:
```diff
@@ -0,0 +1,28 @@
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+
+  outputs = { self, nixpkgs }: let
+      system = "x86_64-linux";
```
it would be nice if other systems are support like darwin and arm, maybe use flake utils?

tami5 commented on lua/sqlite/defs.lua:
```diff
@@ -6,14 +6,13 @@ local M = {}
 --- Load clib
 local clib = (function()
   local path, _
+  path, _ = luv.os_getenv "LIBSQLITE"
```
should only read if vim.g.sqlite_clib_path is not set or is not vim, feels unnecessary call no?

teto replied:
I moved it beneath as a fallback if im.g.sqlite_clib_path is not set

tami5 reviewed (changes requested):
I'm seeing luacheck and style lua still failing but for absolute nonsense it should have been warning instead of errors

tami5 commented on sqlite.lua-scm-1.rockspec:
```diff
@@ -0,0 +1,37 @@
+  tag = "master"
+}
+dependencies = {
+  "plenary.nvim"
```
Oh wait, sqlite.lua doesn't depend on plenary only on luv, have you seen script/gen_rockspec.lua

teto replied:
I think I had seen it then forgot about it when I resumed the PR xD I've submitted an update that adds plenary as a test dependency though. It won't work because plenary is not in the root manifest of luarocks.org but I think this can be done with the help of @tjdevries

teto replied:
```
luarocks test ./sqlite-master-0.rockspec 
Missing dependencies for sqlite master-0:
   plenary.nvim (not installed)

sqlite master-0 depends on plenary.nvim (not installed)
Installing https://luarocks.org/plenary.nvim-scm-1.rockspec
```
fails . 
adding my manifest fixes it (https://luarocks.org/modules/teto/plenary.nvim) but it is not scalable so I wonder if I should manage plenary myself or if @tjdevries has enough time/interest to do it (or another maintainer) .

I think it's fine to merge as is as `luarocks install ./sqlite-master-0.rockspec  --tree /tmp` keeps working

teto replied:
I've just pushed a linting fix. I think it would be best to have the CI use the flake to run the code rather than CI actions. We would have the same results locally as on github + it would work on other CI/CD pipelines without limiting oneself to github. but that's a suggestion.

tami5 replied:
Yes, the CI setup was pre getting into nix. this would make things a lot simpler and won't have this out sync issue.
